src/app/widget_loader.cpp

```diff
@@ -203,7 +203,13 @@ Widget* WidgetLoader::convertXmlElementToWidget(const TiXmlElement* elem, Widget
         // Automatic bind <check> widget with bool preference option
         if (pref) {
           auto prefWidget = new BoolPrefWidget<CheckBox>("");
-          prefWidget->setPref(pref);
+          try {
+            prefWidget->setPref(pref);

```

Comment:
In this case the `setPref` can throw an exception but we would prefer to handle these cases with smart pointers/unique_ptr instead of using throw/catch.

Reply:
I agree it's better to use smart pointers to avoid these kinds of memory leaks. This PR tries to fix the memory leak in a relatively simple way (remain the same semantics). It would be better If those raw pointers are to be replaced with smart pointers in future.

Reply:
I've pushed a fix with a smart pointer for this case: https://github.com/aseprite/aseprite/commit/6157b004e40aa0fbe954b5b6c1539c3b0be37435